grails-app/services/de/iteratec/osm/measurement/environment/BrowserService.groovy

```diff
@@ -27,33 +28,32 @@ class BrowserService {
         return Collections.unmodifiableSet(result)
     }
 
-    List<Browser> findAllByNameOrAlias(List<String> browserNameOrAlias) {
+    List<Browser> findAllByNameOrCreate(List<String> browserNames, boolean isHealthCheck = true) {
```

Comment:
Please separate the logic into two different methods. Having a method that only does what it's called when an flag-parameter is set (and passed into two further levels of detail) is not good practice.
You could have a method `findAllByName` which is used if the browser shouldn't get created. Another method `findOrCreateByName` can call `findAllByName` and create the entity if the result is `null`.